### PR TITLE
use go1.14, mirroring main -collector repo.

### PR DESCRIPTION
**Pull request by lizthegrey**
**Description:**
Updates to go1.14 and circleci's cimg images.

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector/pull/661

**Testing:**
CircleCI should have us covered here.

**Documentation:**
No docs needed.

### CODE REVIEW

File: exporter/awsxrayexporter/go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsxrayexporter
 
-go 1.12
+go 1.14
```
Comment (pjanotti) [inline exporter/awsxrayexporter/go.mod]: hum every time we update we have to change those, I'm thinking that a better practice for this repo will be to remove the version from the module.

Comment (pjanotti) [reply exporter/awsxrayexporter/go.mod]: we can do it separately to not confuse things

File: testbed/tests/metric_test.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ func TestMetric10kDPS(t *testing.T) {
 			testbed.NewOCDataReceiver(testbed.GetAvailablePort(t)),
 			testbed.ResourceSpec{
 				ExpectedMaxCPU: 29,
-				ExpectedMaxRAM: 65,
+				ExpectedMaxRAM: 70,
```
Comment (pjanotti) [inline testbed/tests/metric_test.go]: I assume that there was a test failure for these.

**Review by pjanotti — APPROVED** [review]
LGTM

File: testbed/tests/metric_test.go (hunk 2 above)
Comment (lizthegrey) [reply testbed/tests/metric_test.go]: Affirm, there was.

File: exporter/awsxrayexporter/go.mod (hunk 1 above)
Comment (lizthegrey) [reply exporter/awsxrayexporter/go.mod]: +1 to do it separately. My understanding though is if you don't add it, `go mod tidy` will.

Comment (owais) [reply exporter/awsxrayexporter/go.mod]: I'm not entirely sure but I think we don't strictly need to keep updating this. This sort of refers to the minimum Go version that can build this project and AFAIK, the compiler does not enforce anything using this directive. AFAIK,  it is only used to provide extra information/hints to users when a build fails.

**Pull request merged**